lib/matplotlib/backends/backend_svg.py

```diff
@@ -877,27 +877,37 @@ def draw_image(self, gc, x, y, im, transform=None):
         if clipid is not None:
             self.writer.end('g')
 
+    def _update_glyph_map_defs(self, glyph_map_new):
+        """
+        Emit definitions for not-yet-defined glyphs, and record them as having
+        been defined.
+        """
+        writer = self.writer
+        if glyph_map_new:
+            writer.start('defs')
+            for char_id, (codes, vertices) in glyph_map_new.items():
+                char_id = self._adjust_char_id(char_id)
+                path_data = self._convert_path(
+                    Path(codes, vertices), simplify=False)
+                writer.element('path', id=char_id, d=path_data)
+            writer.end('defs')
+        self._glyph_map.update(glyph_map_new)
```

Comment:
In the original code, the updated was within `if glyph_map_new:`. Is this change intentional?

Reply:
Yes: update()ing a dict with an empty dict doesn't change it, so we can always do it.

Reply:
But why should we? IMHO it's clearer to not call update if we know it does not have any effect. Any here, the code is not even getting more complex for the distinction.

Reply:
I think it's simpler to have less code under the branch, but I don't really care enough.  Moved it into the branch.